src/Tracker.cpp

```diff
@@ -487,8 +497,68 @@ void Tracker::TrackFrame(ImageBWMap& imFrames, ros::Time timestamp, bool bDraw)
       ROS_DEBUG_STREAM("About to test neednew, mkf depth: " << mpCurrentMKF->mdTotalDepthMean);
 
       static GVars3::gvar3<int> gvnAddingMKFs("AddingMKFs", 1, GVars3::HIDDEN | GVars3::SILENT);
-      // Heuristics to check if a key-frame should be added to the map:
-      if (mbAddNext ||  // mMapMaker.Initializing() ||
+
+      bool use_entropy_keyframe = true;
+      if(use_entropy_keyframe) // use the entropy based keyframe method (CPER)
+      {
+          TooN::Vector<3> trackerEntropy = EvaluateTracker(this);
+          //std::cout<<"tracker entropy: " << trackerEntropy <<std::endl;
+          ROS_INFO("tracker entropy: (%f,%f,%f)",trackerEntropy[0],trackerEntropy[1],trackerEntropy[2]);
+          double entropyThresh = -4.0; //todo (adas) pull this out to a param
+          bool addEntropyMKF = false;
+          RecordMeasurementsAndBufferKeyFrame(); //todo (adas) turn into circular buffer
+
+          //MultiKeyFrame* closest_mkf = mMapMaker.ClosestMultiKeyFrame(*mpCurrentMKF);
+          //double distance_to_closest_mkf = closest_mkf->Distance(*mpCurrentMKF);
+          //ROS_INFO_STREAM("distance to closest mkf is: " << distance_to_closest_mkf);
+
+          //bool test = mMapMaker.HasLocalMapConverged();
+          int tracker_queue_size = mMapMaker.TrackerQueueSize();
+          //ROS_WARN_STREAM("q size: " <<test_size);
+
+          if( (trackerEntropy[0] > entropyThresh ) || (trackerEntropy[1] > entropyThresh ) || (trackerEntropy[2] > entropyThresh ) )
+          {
+          		if(mMap.mbFreshMap)
+          			addEntropyMKF = true;
+          		else
+          		{
+          			ROS_WARN_STREAM("Waiting for fresh map...");
+          			addEntropyMKF = false;
+          		}
+          } //todo (adas) include rotational entropies as well, plus options for the threshold type (just trans, just rot, both, etc)
+
+          // && tracker_queue_size < 1 && (ros::Time::now() - mtLastMultiKeyFrameDropped > ros::Duration(0.2))
+
+          if(addEntropyMKF ) //add new keyframe //todo (adas) figure how to to limit the number of mkf we add at once.  Problem is that we can't tell how long it will take for the mkf to get into the map.  Need a smart way of adding a "delay" between additions
+          {
+              mMessageForUser << " SHOULD BE Adding MultiKeyFrame to Map";
+              //sort keyframe vec
+              std::sort (vKeyframeScores.begin(), vKeyframeScores.end(), kfComparitor);
+
+              if(vKeyframeScores.size()>0)
+              {
+                  double bestSortIndex = vKeyframeScores[0].second;
+                  //int k_itr = 1;
+                  //while( bestSortIndex < vKeyframeScores.size())
+                  //{
+                  //    bestSortIndex = vKeyframeScores[k_itr].second;
+                  //    k_itr++;
+                  //} //todo (adas) get rid of this, turn into circular buffer, this is a hack!
```

Comment:
These lines are not present in current code (Stan probably took care of it). @arunxdas Could you please double-check if you need them. I'll open an issue.